src/main/java/chatbot/util/Parser.java

```diff
@@ -0,0 +1,201 @@
+package chatbot.util;
+
+import chatbot.datetime.Timestamp;
+import chatbot.exception.ChatBotException;
+
+
+/**
+ * Represents an interpreter for user inputs.
+ */
+public class Parser {
+
+    private final Ui ui;
+    private final Storage storage;
+    private final TaskList taskList;
+
+    /**
+     * Instantiates a new Parser.
+     *
+     * @param ui The UI.
+     * @param storage   The storage.
```

Comment:
Perhaps comments can be more descriptive of what UI/storage/tasklist is or contains? :)